commit fc07d9caf22ee609c03b2edd5ca03b17fd5c2098
Author: Michael Palimaka <kensington@gentoo.org>
Date:   Wed May 13 04:22:45 2015 +1000

    Fix build with >=cmake-3.2.

    The FindOpenGL module no longer pulls in X11 libraries.

--- source/shared_lib/CMakeLists.txt
+++ source/shared_lib/CMakeLists.txt
@@ -95,6 +95,10 @@ IF(BUILD_MEGAGLEST_MODEL_VIEWER OR BUILD_MEGAGLEST_MAP_EDITOR OR BUILD_MEGAGLEST
 		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${OPENGL_LIBRARY})
 	ENDIF()
 
+        IF(UNIX)
+		FIND_PACKAGE(X11 REQUIRED)
+		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${X11_LIBRARIES})
+        ENDIF()
 
         IF(WANT_XERCES)
 	        FIND_PACKAGE(XercesC REQUIRED)
